### PR TITLE
Update libqmi to 1.26.6 and make QMI-over-MBIM support optional

### PR DESCRIPTION
**Pull request by aleksander0m**
Maintainer: @nickberry17 

Description: Upgrade library version to 1.26.6 and also add a new CONFIG_LIBQMI_WITH_MBIM_QMUX that allows disabling the libmbim dependency altogether (i.e. for systems running with modems in QMI mode explicitly).

### CODE REVIEW

**Comment by aleksander0m** [comment]
I've ignored the optional QRTR support for now, as that is not fully stable yet.

File: libs/libqmi/Makefile [hunk 1]
```diff
@@ -9,7 +9,7 @@ include $(TOPDIR)/rules.mk
 
 PKG_NAME:=libqmi
 PKG_VERSION:=1.26.6
-PKG_RELEASE:=1
+PKG_RELEASE:=2
```
Comment (neheb) [inline libs/libqmi/Makefile:12]: If you're already updating the package with this change, why bump PKG_RELEASE?

File: libs/libqmi/Makefile [hunk 2]
```diff
@@ -59,11 +65,16 @@ CONFIGURE_ARGS += \
 	--disable-gtk-doc-pdf \
 	--disable-silent-rules \
 	--enable-firmware-update \
-	--enable-mbim-qmux \
```
Comment (neheb) [inline libs/libqmi/Makefile:62]: Could do

```
--$(if $(LIBQMI_WITH_MBIM_QMUX),en,dis)able-mbim-qmux \
```

File: libs/libqmi/Makefile (hunk 1 above)
Comment (nickberry17) [reply libs/libqmi/Makefile:12]: A very welcome change, @aleksander0m !  Thanks for your hard work.  Correct, if the PKG_VERSION increases, then PKG_RELEASE resets to 1.

Comment (aleksander0m) [reply libs/libqmi/Makefile:12]: @neheb @nickberry17 the thing is that this PR has 3 commits, one doing `1.26.2-1 -> 1.26.4-1`; another one doing `1.26.4-1 -> 1.26.6-1`; and the last one adding the new config option, bumping PKG_RELEASE: `1.26.6-1 -> 1.26.6-2`. I didn't want to have the config change in the same commit as the version bump, as it was a completely orthogonal thing. Would it be better if I just send one single commit with everything? I personally prefer the 3 separate commits, but whatever you prefer.

File: libs/libqmi/Makefile (hunk 2 above)
Comment (aleksander0m) [reply libs/libqmi/Makefile:62]: oh, nice, will change that

File: libs/libqmi/Makefile (hunk 1 above)
Comment (neheb) [reply libs/libqmi/Makefile:12]: I mean up to you. Typically, a PR either bumps the PKG_RELEASE or resets it to 1 as anything intermediate is not present.

Comment (aleksander0m) [reply libs/libqmi/Makefile:12]: If it's up to me, I'd prefer to leave it as 3 separate commits, each bumping only the version field that needs to be bumped. In my mind a PR is a collection of commits, but each commit needs to be self-contained by itself. Makes it easier e.g. to cherry-pick to the 19.07 branch if we want to cherry-pick the version bump only and not the new config change.

**Pull request merged**